src/deploy/osp_deployer/profile.py

```diff
@@ -0,0 +1,120 @@
+        self.profile_name = self.settings.profile
+        self.definition = self.settings.foreman_configuration_scripts + \
+            '/deploy/osp_deployer/profiles/' + self.profile_name + ".json"
+        if self.profile_name == "custom":
+            logger.info("Using Custom profile")
+            return
+        if not os.path.isfile(self.definition):
+            raise AssertionError(self.profile_name +
+                                 " is not a valid " +
+                                 " profile in your .ini settings")
+
+    def validate_configuration(self):
+        # Make sure the user ini matches the associated profile settings
+        if self.profile_name == "custom":
+            return
+        with open(self.definition) as profile_config:
+            profile_definition = json.load(profile_config)
+        user_config = ConfigParser()
+        user_config.read(self.settings.settings_file)
+        profile_config = ConfigParser()
+        profile_config.read(os.path.dirname(inspect.getfile(Settings)) +
+                            "/" + profile_definition['sample_ini'])
+        Err = ''
+        for items in profile_definition['associated_settings']:
+            for stanza, value in items.iteritems():
+                for set, vals in value.iteritems():
+                    allowed_settings = []
+                    validated = False
+                    try:
+                        if vals['valid_values']:
+                            allowed_settings = vals['valid_values']
+                    except:
+                        pass
+                    try:
+                        if vals['validate']:
+                            for test in vals['validate']:
+                                if 'should_be_valid_ip' in test:
+                                    if self.is_valid_ip(user_config.get(str(stanza), set)) is False:
+                                        Err = Err + "\nSetting for " + set + \
+                                              " Should be a valid ip adress\n"
+                            validated = True
+                    except:
+                        pass
+                    if len(allowed_settings) > 0:
+                        if user_config.get(str(stanza), set) in allowed_settings:
+                            pass
+                        else:
+                            Err = Err + "\nYour setting for " + stanza + \
+                                  "::" + set + \
+                                  " is not valid for the profile " + \
+                                  self.profile_name + \
+                                  " it should be to one of the following " + \
+                                  "options : " + \
+                                  json.dumps(allowed_settings)
+                    elif validated is True:
+                        pass
+                    else:
+                        usr_sett = != user_config.get(str(stanza), set)
```

Comment:
Looks like a syntax error on the above line? "= !="

Reply:
thanks, fixed